example/src/main/java/io/netty/example/tunechoserver/TunEchoServer.java

```diff
@@ -0,0 +1,452 @@
+
+        try {
+            _threadPoolSize = Integer.parseInt(System.getProperty("thread-pool-size", "0"));
+            if (_threadPoolSize < 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            System.err.println("Invalid value specified for thread-pool-size property: " +
+                    System.getProperty("thread-pool-size"));
+            return false;
+        }
+
+        try {
+            _readBufSize = Integer.parseInt(System.getProperty("read-buf-size", "2048"));
+            if (_readBufSize < 1) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            System.err.println("Invalid value specified for read-buf-size property: " +
+                    System.getProperty("read-buf-size"));
+            return false;
+        }
+
+        if (Boolean.parseBoolean(System.getProperty("use-unpooled", "false"))) {
+            _bufAllocator = UnpooledByteBufAllocator.DEFAULT;
+        }
+
+        _forceGC = Boolean.parseBoolean(System.getProperty("force-gc", "false"));
+
+        _debug = Boolean.parseBoolean(System.getProperty("debug", "true"));
+
+        return true;
+    }
+
+    public static void main(String[] args) {
+
+        if (!readProps()) {
+            return;
+        }
+
+        System.out.println("TunEchoServer starting");
+
+        try {
+            _logger = Logger.getLogger("");
+            _logger.setLevel(_debug ? Level.FINEST : Level.SEVERE);
+
+            if (_threadPoolSize > 0) {
+                _threadPool = (ThreadPoolExecutor) Executors.newFixedThreadPool(_threadPoolSize);
+            }
+
+            final TunEchoServer server = new TunEchoServer();
+
+            EventLoopGroup group = new EpollEventLoopGroup(1);
+
+            ChannelFactory<Channel> channelFactory = new ChannelFactory<Channel>() {
+                @Override
+                public Channel newChannel() {
+                    try {
+                        Channel channel = new TunTapChannel();
+                        if (_debug) {
+                            channel.pipeline().addLast(new TunTapPacketLogger());
+                        }
+                        return channel;
+                    } catch (Throwable t) {
+                        throw new ChannelException("Unable to create TunTapChannel", t);
+                    }
+                }
+            };
+
+            Bootstrap bootstrap = new Bootstrap()
+                .group(group)
+                .channelFactory(channelFactory)
+                .option(ChannelOption.ALLOCATOR, _bufAllocator)
+                .option(TunTapChannelOption.READ_BUF_SIZE, _readBufSize)
+                .handler(server);
+
+            final TunTapChannel channel =
+                    (TunTapChannel) bootstrap.bind(new TunAddress(_tunDeviceName)).sync().channel();
+
+            System.out.println("TunEchoServer ready");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS106)